rust/otap-dataflow/crates/otlp/build.rs

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! A build script to generate the gRPC OTLP receiver API (client and server stubs.
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // To regenerate the gRPC API from the proto file:
+    // - Uncomment the following lines.
+    // - Run `cargo build` to regenerate the API.
+    // - Comment the following lines.
+    // - Commit the changes.
+    // tonic_build::configure()
+    //     .out_dir("src/grpc/grpc_stubs")
+    //     .compile_protos(
+    //         &[
+    //             "../../../../proto/opentelemetry/proto/collector/logs/v1/logs_service.proto",
+    //             "../../../../proto/opentelemetry/proto/collector/metrics/v1/metrics_service.proto",
+    //             "../../../../proto/opentelemetry/proto/collector/trace/v1/trace_service.proto",
+    //         ],
+    //         &["../../../../proto"],
+    //     )?;
+    // Ok(())
+}
```

Comment:
Since we vendor the files generated by `protoc` in the repo, I suggest moving the generation logic into a new xtask called `protoc`, which would roughly contain what’s currently in build.rs. This will later be extended with what Joshua is working on in the experimental repo. @jmacd I'd like to get your feedback on this.